src/tuvx/grid_map.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright (C) 2023-2024 National Center for Atmospheric Research
+// SPDX-License-Identifier: Apache-2.0
+//
+// This file contains the implementation of the TUVX class, which represents a multi-component
+// reactive transport model. It also includes functions for creating and deleting TUVX instances.
+#include <musica/tuvx/grid_map.hpp>
+
+#include <cstring>
+#include <filesystem>
+#include <iostream>
+
+namespace musica
+{
+
+  // GridMap external C API functions
+
+  GridMap *CreateGridMap(Error *error)
+  {
+    DeleteError(error);
+    return new GridMap(error);
+  }
+    
+  void DeleteGridMap(GridMap *grid_map, Error *error)
+  {
+    DeleteError(error);
+    try
+    {
+      delete grid_map;
+    }
+    catch (const std::system_error &e)
+    {
+      *error = ToError(e);
+    }
+    *error = NoError();
+  }
+
+  void AddGrid(GridMap *grid_map, Grid *grid, Error *error)
+  {
+    DeleteError(error);
+    grid_map->AddGrid(grid, error);
+  }
+
+  Grid *GetGrid(GridMap *grid_map, const char *grid_name, const char *grid_units, Error *error)
+  {
+    DeleteError(error);
+    return grid_map->GetGrid(grid_name, grid_units, error);
+  }
+
+  // GridMap class functions
+
+  GridMap::GridMap(Error *error)
+  {
+    int error_code = 0;
+    grid_map_ = InternalCreateGridMap(&error_code);
+    if (error_code != 0)
+    {
+      *error = Error{ 1, CreateString(MUSICA_ERROR_CATEGORY), CreateString("Failed to create grid map") };
+    }
+    owns_grid_map_ = true;
+    *error = NoError();
+  }
+
+  GridMap::~GridMap()
+  {
+    int error_code = 0;
+    if (grid_map_ != nullptr && owns_grid_map_)
+    {
+      InternalDeleteGridMap(grid_map_, &error_code);
+    }
+    grid_map_ = nullptr;
+    owns_grid_map_ = false;
```

Comment:
```suggestion
    grid_map_ = nullptr;
    owns_grid_map_ = false;
```

Is `owns_grid_map_` the same thing as `grid_map_` being null?

Reply:
good point. I think they were needed in an earlier iteration, but I agree they seem to be unnecessary now. will remove

Reply:
for the map classes (`GridMap` and `ProfileMap`) the `owns_*` flags are needed because they can reference maps owned by the TUV-x instance, but for `Grid` and `Profile` they aren't needed and have been removed.